api/secrets/models/secret.go

```diff
@@ -0,0 +1,57 @@
+package models
+
+// Secret holds general information about secret
+// swagger:model Secret
+type Secret struct {
+	// Name of the secret or its property, related to type and resource)
+	//
+	// required: true
+	// example: db_password
+	Name string `json:"name"`
+
+	// DisplayName of the secret
+	//
+	// required: false
+	// example: Database password
+	DisplayName string `json:"displayName,omitempty"`
+
+	// Type of the secret
+	//
+	// required: false
+	// example: csi-az-blob
+	Type SecretType `json:"type,omitempty"`
+
+	// Resource of the secrets
+	//
+	// required: false
+	// example: volumeAbc
+	Resource string `json:"resource,omitempty"`
+
+	// Component name of the component having the secret
+	//
+	// required: false
+	// example: api
+	Component string `json:"component,omitempty"`
+
+	// Status of the secret
+	// - Pending = Secret exists in Radix config, but not in cluster
+	// - Consistent = Secret exists in Radix config and in cluster
+	// - Orphan = Secret does not exist in Radix config, but exists in cluster
+	//
+	// required: false
+	// example: Consistent
+	Status string `json:"status,omitempty"`
+}
+
+type SecretType string
+
+const (
+	SecretTypeGeneric               SecretType = "generic"
+	SecretTypeClientCert            SecretType = "client-cert"
+	SecretTypeAzureBlobFuseVolume   SecretType = "azure-blob-fuse-volume"
+	SecretTypeCsiAzureBlobVolume    SecretType = "csi-azure-blob-volume"
+	SecretTypeCsiAzureKeyVaultCreds SecretType = "csi-azure-key-vault-creds"
+	SecretTypeCsiAzureKeyVaultItem  SecretType = "csi-azure-key-vault-item"
+	SecretTypeOrphaned              SecretType = "orphaned"
```

Comment:
Why do we define secret types SecretTypeOrphaned and SecretTypePendig. Isn't Pending and Orhapned a Status?

Reply:
Good point. Then we need to apply to it a logic, recognizing the type from labels and names I guess. I just quickly added this type to loops for handling orphan and pending secrets.

Reply:
Ok. I guess it is good for now then